packages/orchestration/src/orchestration-utils.ts

```diff
@@ -0,0 +1,155 @@
+import { createLogger } from '@sap-cloud-sdk/util';
+import type {
+  CompletionPostRequest,
+  FilteringStreamOptions,
+  ModuleConfigs,
+  OrchestrationConfig,
+  OutputFilteringConfig
+} from './client/api/schema/index.js';
+import type {
+  Prompt,
+  StreamOptions,
+  LlmModuleConfig,
+  OrchestrationModuleConfig
+} from './orchestration-types.js';
+
+const logger = createLogger({ messageContext: 'orchestration-utils' });
+
+/**
+ * @internal
+ */
+export function constructCompletionPostRequestFromJsonModuleConfig(
+  config: Record<string, any>,
+  prompt?: Prompt,
+  stream?: boolean
+): Record<string, any> {
+  return {
+    messages_history: prompt?.messagesHistory || [],
+    input_params: prompt?.inputParams || {},
+    orchestration_config: stream
+      ? { ...config, stream: true }
+      : { ...config, stream: false }
+  };
+}
+
+/**
+ * @internal
+ */
+export function addStreamOptionsToLlmModuleConfig(
+  llmModuleConfig: LlmModuleConfig,
+  streamOptions?: StreamOptions
+): LlmModuleConfig {
+  if (!streamOptions?.llm) {
+    return llmModuleConfig;
+  }
+  return {
+    ...llmModuleConfig,
+    model_params: {
+      ...llmModuleConfig.model_params,
+      stream_options: {
+        include_usage: true,
+        ...(llmModuleConfig.model_params.stream_options || {}),
+        ...streamOptions.llm
+      }
+    }
+  };
+}
+
+/**
+ * @internal
+ */
+export function addStreamOptionsToOutputFilteringConfig(
+  outputFilteringConfig: OutputFilteringConfig,
+  filteringStreamOptions: FilteringStreamOptions
+): OutputFilteringConfig {
+  return {
+    ...outputFilteringConfig,
+    stream_options: {
+      ...(outputFilteringConfig.stream_options || {}),
+      ...filteringStreamOptions
+    }
+  };
+}
+
+/**
+ * @internal
+ */
+export function addStreamOptions(
+  moduleConfigs: ModuleConfigs,
+  streamOptions?: StreamOptions
+): OrchestrationConfig {
+  const { llm_module_config, filtering_module_config } = moduleConfigs;
+  const outputFiltering = streamOptions?.outputFiltering;
+  const chunkSize = streamOptions?.chunk_size;
+
+  if (!moduleConfigs?.filtering_module_config?.output && outputFiltering) {
+    logger.warn(
+      'Output filter stream options are not applied because filtering module is not configured.'
+    );
+  }
+
+  return {
+    stream: true,
+    stream_options: {
+      chunk_size: chunkSize
```

Comment:
See my previous comment, would be nice if we can do `stream_options: ...streamOptions?.global`